CMakeLists.txt

```diff
@@ -70,7 +71,12 @@ endif()
 if (DEFINED PYTHON_INCLUDE_DIR AND DEFINED PYTHON_LIBRARIES AND DEFINED Python_EXECUTABLE)
     SET(PYTHON_INCLUDE_DIRS "${PYTHON_INCLUDE_DIR}")
     SET(PYTHONLIBS_FOUND TRUE)
-    find_package(Python 2.7 EXACT REQUIRED COMPONENTS Interpreter)
+    # Use the Python module to find the python lib.
+    if(BUILD_WITH_PYTHON_3)
+        find_package(Python 3.7 EXACT REQUIRED COMPONENTS Interpreter)
+    else()
+        find_package(Python 2.7 EXACT REQUIRED COMPONENTS Interpreter)
+    endif()
     if(NOT Python_Interpreter_FOUND)
         set(PYTHONLIBS_FOUND FALSE)
     endif()
```

Comment:
I noticed that we've got `find_package()` calls both here and in `cmake/python.cmake`. Do we need all of this here? Could everything from lines 71-86 be replaced with simply `include(cmake/python.cmake)`? Or could it at least all be moved into `cmake/python.cmake` to avoid the redundancy?

I'd be ok with it going in as is for now, but it'd be nice to see that in a follow-up PR.

Reply:
This block here from lines 71-76 is for our internal Maya build where we build maya-usd inside. We are passing in the 3 python vars (line 71) and then calling find_package() which doesn't really do anything since all the vars were set. Unless something bad happens in which case we reset (line 75). We cannot remove this, but it could probably be moved inside of cmake/python.cmake.

Reply:
Good point @mattyjams , I am planning to revisit python.cmake and some of the logic here. There is definitely some redundancy here but let's leave it for another PR.